File: data-structure/java/arrays/Stack.java
public class Stack {
    private int capacity;
    private int top = -1;
    private int values[];

    public Stack(int capacity) {
        this.capacity = capacity;
        this.values = new int[capacity];
    }

    public void print() {
        for (int i = 0; i <= this.top; i++) {
            System.out.println(i + " -- " + this.values[i]);
        }
    }

    public void push(int value) {
        if (this.top == this.capacity - 1) {
            System.out.println("The array is already filled");
            return;
        }

        this.top += 1;
        this.values[this.top] = value;
    }

    public void pop() {
        if (this.top == -1) {
            System.out.println("The array is empty");
            return;
        }

        this.top -= 1;
    }

    public int getTop() {
        return this.values[this.top];
    }

    public static void main(String[] args) {
        Stack s = new Stack(5);
        s.push(5);
        s.push(3);
        s.push(1);
        s.push(4);
        s.push(2);
        s.print();
        s.pop();
        s.pop();
        s.pop();
        s.pop();
        s.pop();
        s.print();
        s.push(6);
        System.out.println(s.getTop());
    }
}

File: data-structure/js/arrays/stack.ts
class Stack {
  private capacity: number;
  private top: number;
  private values: number[];
  constructor(capacity: number) {
    this.values = new Array(capacity);
    this.capacity = capacity;
    this.top = -1;
  }

  filled() {
    return this.top === this.capacity - 1;
  }

  empty() {
    return this.top === -1;
  }

  stack(value: number) {
    if (this.filled()) console.log("The stack is already filled");
    else {
      this.top++;
      this.values[this.top] = value;
    }
  }

  unstack() {
    if (this.empty()) console.log("The stack is empty");
    else {
      this.top--;
    }
  }

  getTop() {
    if (this.empty()) console.log("The stack is empty");
    else {
      return this.values[this.top];
    }
  }
}

const s = new Stack(5);
s.getTop();
s.stack(1);
s.stack(2);
s.stack(3);
s.stack(4);
s.stack(5);
s.unstack();
s.stack(6);
console.log(s.getTop());
s.unstack();
s.unstack();
s.unstack();
s.unstack();
s.unstack();
s.unstack();
